### PR TITLE
Add `hermitian()` to matrix gallery

### PR DESCRIPTION
**Pull request by swaptr**
## Description
Added `hermitian()` to generate random hermitian matrices of size `n`.

Issue/s resolved: #1073 

## Changes proposed:
- Add `hermitian()` to `utils.data.matrixgallery`.

## Type of change
- New feature (non-breaking change which adds functionality)

## Memory requirements
- NA

## Performance
- NA

## Due Diligence
- [ ] All split configurations tested
- [ ] Multiple dtypes tested in relevant functions
- [ ] Documentation updated (if needed)
- [ ] Title of PR is suitable for corresponding CHANGELOG entry

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

**Comment by swaptr** [comment]
@ClaudiaComito I could'nt figure out the split logic for this. Can you please review the PR?

File: heat/utils/data/matrixgallery.py [hunk 1]
```diff
@@ -1,153 +1,187 @@
...
+
+__all__ = ["hermitian", "parter", "random_known_singularvalues", "random_known_rank"]
+
+
+def hermitian(
+    n: int,
+    split: Union[None, int] = None,
+    device: Union[None, str, Device] = None,
+    comm: Union[None, Communication] = None,
+    dtype: Type[datatype] = core.float32,
```
Comment (ClaudiaComito) [inline heat/utils/data/matrixgallery.py:24]:
```suggestion
    dtype: Type[datatype] = core.complex64,
```

File: heat/utils/data/matrixgallery.py [hunk 2]
```diff
@@ -1,153 +1,187 @@
...
+) -> DNDarray:
+    """
+    Generates a Hermitian matrix of arbitrary size. A Hermitian matrix is a square matrix that is equal to the transpose of its
+    conjugate matrix.
```
Comment (ClaudiaComito) [inline heat/utils/data/matrixgallery.py:28]:
```suggestion
    Generates a Hermitian matrix of arbitrary size. A Hermitian matrix is a  complex square matrix that is equal to its conjugate transpose.
```

File: heat/utils/data/matrixgallery.py [hunk 3]
```diff
@@ -1,153 +1,187 @@
...
+        size of the resulting square matrix
+    split: None or int, optional
+        The axis along which the array content is split and distributed in memory.
+    device: None or str or Device, optional
+        Specifies the device the tensor shall be allocated on, defaults globally set default device.
+    comm : Communication, optional
+        Handle to the nodes holding distributed parts or copies of this array.
+    dtype: Type[datatype], optional
+        The desired data-type for the array, defaults to ht.float64.
+
```
Comment (ClaudiaComito) [inline heat/utils/data/matrixgallery.py:42]:
```suggestion
    dtype: Type[datatype], optional
        The desired data-type for the array, defaults to ht.complex64.

```

File: heat/utils/data/matrixgallery.py [hunk 4]
```diff
@@ -1,153 +1,187 @@
...
+    References
+    ----------
+    [1] https://en.wikipedia.org/wiki/Hermitian_matrix
+    """
+    matrix = core.arange(n * n, dtype=dtype, device=device, comm=comm).reshape(n, n)
+    matrix += 1j * core.arange(n * n).reshape(n, n)
+    matrix = matrix + core.conj(matrix.T)
+    return matrix
+
```
Comment (ClaudiaComito) [inline heat/utils/data/matrixgallery.py:51]:
```suggestion
    return matrix @ core.conj(matrix).T
```

**Review by ClaudiaComito — CHANGES_REQUESTED** [review]
Thanks again @swaptr and apologies for the delay. I suggest to use `ht.random.randn` instead of `arange`, and that also allows specifying a split axis.

File: heat/utils/data/matrixgallery.py [hunk 5]
```diff
@@ -1,153 +1,186 @@
...
+    comm : Communication, optional
+        Handle to the nodes holding distributed parts or copies of this array.
+    dtype: Type[datatype], optional
+        The desired data-type for the array, defaults to ht.complex64.
+
+    References
+    ----------
+    [1] https://en.wikipedia.org/wiki/Hermitian_matrix
+    """
+    real_dtype = types.float32 if dtype is types.complex64 else types.float64
```
Comment (ClaudiaComito) [inline heat/utils/data/matrixgallery.py:46]:
```suggestion
    real_dtype = core.float32 if dtype is core.complex64 else core.float64
```

**Review by ClaudiaComito — CHANGES_REQUESTED** [review]

**Pull request merged**